htdocs/core/class/notify.class.php

```diff
@@ -651,15 +633,11 @@ public function send($notifcode, $object, $filename_list = array(), $mimetype_li
 						}
 
 						$ref = dol_sanitizeFileName($newref);
-						$pdf_path = $dir_output."/".$ref.".pdf";
-						if (!dol_is_file($pdf_path)||(is_object($arraydefaultmessage) && $arraydefaultmessage->id > 0 && !$arraydefaultmessage->joinfiles)) {
-							// We can't add PDF as it is not generated yet.
-							$filepdf = '';
-						} else {
-							$filepdf = $pdf_path;
+						if (is_object($arraydefaultmessage) && $arraydefaultmessage->id > 0 && !$arraydefaultmessage->joinfiles) {
+							$filepdf = (dol_most_recent_file($dir_output, preg_quote($ref, '/').'[^\-]+'))['fullname'];
```

Comment:
The dol_most_recent_filecreturn the latest file linked to object. It can be any document, a legal private document for example.
On notification, we only want to send document that are the generated document. So document call ref.pdf it it exists.

Reply:
Yes, that's why there's a filter that limits to a list to the only file starting with the ref.
But if it's not limiting enough, how can I proceed?